doc/jvm/xgboost_spark_migration.rst

```diff
@@ -0,0 +1,140 @@
+######################################################
+Migration Guide: How to migrate from XGBoost Spark 3.x
+######################################################
+
+XGBoost Spark underwent significant modifications beginning with version 3.0,
+which may cause compatibility issues with existing user code.
+
+This guide will walk you through the process of updating your code to ensure
+it's compatible with XGBoost Spark 3.0 and later versions.
+
+**********************
+XGBoost Spark Packages
+**********************
+
+XGBoost Spark 3.0 introduced a single uber package named xgboost-spark_2.12-3.0.0.jar, which bundles 
```

Comment:
not entirely sure whether it's necessary to be an Uber package or the implication of this.

Reply:
I think we have resolved this confusion.